src/expr/src/scalar/func.rs

```diff
@@ -2040,34 +2040,16 @@ where
 }
 
 fn date_trunc_interval<'a>(a: Datum, int: Interval) -> Result<Datum<'a>, EvalError> {
-    use mz_repr::adt::datetime::DateTimeField;
+    let mut int2 = int.clone();
```

Comment:
You can declare `int` mutable in the function declaration, no need to clone here.

Reply:
Hm. I can put the mut in the declaration, but then how do I call that function? In line 2539 i have get a datum and that only gives me a non mutable interval (afaik).

Reply:
`int: Interval` is call-by-value, so you get an owned object. Adding `mut` just makes the local object mutable. This is the same as:
```
fn date_trunc_interval<'a>(a: Datum, int: Interval) -> Result<Datum<'a>, EvalError> {
    let mut int = int;
    ...
}
```

Reply:
OK. The compiler told me how this can be done...